web-service/helpers.py

```diff
@@ -37,9 +37,14 @@
 
 ################################################################################
 
-def BuildResponse(objects):
+def BuildResponse(objects, request_headers):
     metadata_output = []
 
+    if 'Origin' in request_headers:
+        base_url = request_headers['Origin']
```

Comment:
Are we sure we prefer to use Origin header?  I thought that is set to the origin of the request, which would not likely have a `/favicon?` endpoint..